### PR TITLE
feat: confidence sequences estimator

### PR DESCRIPTION
**Pull request by bassmang**

### CODE REVIEW

File: vowpalwabbit/core/include/vw/core/confidence_sequence.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      if (lo != 0.0)
+      {
+        this->partials[i] = lo;
+        ++i;
+      }
+      x = hi;
+    }
+    this->partials[i] = x;
+    this->partials.resize(i + 1);
+    return *this;
```
Comment (lalo) [inline vowpalwabbit/core/include/vw/core/confidence_sequence.h:41]: make a test similar to this one distributionally_robust_test.cc

File: vowpalwabbit/core/include/vw/core/confidence_sequence.h [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      double hi = x + y;
+      double lo = y - (hi - x);
+      if (lo != 0.0)
+      {
+        this->partials[i] = lo;
+        ++i;
+      }
+      x = hi;
+    }
+    this->partials[i] = x;
```
Comment (ataymano) [inline vowpalwabbit/core/include/vw/core/confidence_sequence.h:42]: isn't i out of range here?

File: vowpalwabbit/core/include/vw/core/confidence_sequence.h (hunk 1 above)
Comment (bassmang) [reply vowpalwabbit/core/include/vw/core/confidence_sequence.h:45]: Added test

File: vowpalwabbit/core/include/vw/core/confidence_sequence.h (hunk 2 above)
Comment (bassmang) [reply vowpalwabbit/core/include/vw/core/confidence_sequence.h:43]: Good catch, added loop to ensure it is not out of range

**Review by lalo — APPROVED** [review]

File: vowpalwabbit/core/include/vw/core/confidence_sequence.h (hunk 2 above)
Comment (ataymano) [reply vowpalwabbit/core/include/vw/core/confidence_sequence.h:43]: doesn't swapping this line with resizing work here?

Comment (bassmang) [reply vowpalwabbit/core/include/vw/core/confidence_sequence.h]: Yes that should work, updated

**Pull request merged**